sphinx/make_mode.py

```diff
@@ -75,7 +75,11 @@ def build_clean(self):
             return 1
         print("Removing everything under %r..." % self.builddir)
         for item in os.listdir(self.builddir):
-            shutil.rmtree(self.builddir_join(item))
+            itempath = self.builddir_join(item)
+            if path.isdir(itempath):
```

Comment:
I prefer to separate these operations to utility function like `sphinx.util.osutil.rmtree()`.